Marlin/src/module/motion.cpp

```diff
@@ -1436,6 +1452,16 @@ void homeaxis(const AxisEnum axis) {
 
   #endif
 
+  // Put away the Z probe
+  #if HOMING_Z_WITH_PROBE
+    if (axis == Z_AXIS) {
+      #if ENABLED(BLTOUCH)
+        if (set_bltouch_deployed(false)) return;
+      #endif
+      if (STOW_PROBE()) return;
+    }
+  #endif
```

Comment:
Moved this down to make the revert more complete, but if the other reverts solve the Z position problem then this may turn out to be ok moved back up above.

Reply:
Yes this will solve Z. STOW_PROBE move Z